src/Host/Client/Impl/Session/RSession.cs

```diff
@@ -38,7 +37,7 @@ internal sealed class RSession : IRSession, IRCallbacks {
         public event EventHandler<EventArgs> Disconnected;
         public event EventHandler<EventArgs> Disposed;
         public event EventHandler<EventArgs> DirectoryChanged;
-        public event EventHandler<EventArgs> BeforePackagesInstalled;
+        public event AsyncEventHandler<EventArgs> BeforePackagesInstalledAsync;
```

Comment:
Async events have been a bug prone in CPS for years and I don't want to introduce them in release branch. We have `IRSessionCallback` where you can add another method.